crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -0,0 +1,1285 @@
+            .unwrap()
+            .extract::<Option<String>>()?;
+        self.num_qubits = state
+            .get_item("num_qubits")?
+            .unwrap()
+            .extract::<Option<usize>>()?;
+        self.dt = state.get_item("dt")?.unwrap().extract::<Option<f64>>()?;
+        self.granularity = state.get_item("granularity")?.unwrap().extract::<i32>()?;
+        self.min_length = state.get_item("min_length")?.unwrap().extract::<usize>()?;
+        self.pulse_alignment = state
+            .get_item("pulse_alignment")?
+            .unwrap()
+            .extract::<i32>()?;
+        self.acquire_alignment = state
+            .get_item("acquire_alignment")?
+            .unwrap()
+            .extract::<i32>()?;
+        self.qubit_properties = state
+            .get_item("qubit_properties")?
+            .unwrap()
+            .extract::<Option<Vec<PyObject>>>()?;
+        self.concurrent_measurements = state
+            .get_item("concurrent_measurements")?
+            .unwrap()
+            .extract::<Vec<Vec<usize>>>()?;
+        self.gate_map = IndexMap::from_iter(
+            state
+                .get_item("gate_map")?
+                .unwrap()
+                .extract::<GateMapState>()?
+                .into_iter()
+                .map(|(name, prop_map)| (name, IndexMap::from_iter(prop_map.into_iter()))),
+        );
+        self._gate_name_map = state
+            .get_item("gate_name_map")?
+            .unwrap()
+            .extract::<IndexMap<String, TargetOperation, RandomState>>()?;
+        self.global_operations = state
+            .get_item("global_operations")?
+            .unwrap()
+            .extract::<IndexMap<u32, HashSet<String>, RandomState>>()?;
+        self.qarg_gate_map = IndexMap::from_iter(
+            state
+                .get_item("qarg_gate_map")?
+                .unwrap()
+                .extract::<Vec<(Option<Qargs>, Option<HashSet<String>>)>>()?,
+        );
+        self.non_global_basis = state
+            .get_item("non_global_basis")?
+            .unwrap()
+            .extract::<Option<Vec<String>>>()?;
+        self.non_global_strict_basis = state
+            .get_item("non_global_strict_basis")?
+            .unwrap()
+            .extract::<Option<Vec<String>>>()?;
+        Ok(())
+    }
+}
+
+// Rust native methods
+impl Target {
+    /// Returns an iterator over all the instructions present in the `Target`
+    /// as pair of `&TargetOperation` and `Option<&Qargs>`.
+    pub fn instructions(&self) -> impl Iterator<Item = (&TargetOperation, Option<&Qargs>)> {
+        self.gate_map.iter().flat_map(move |(op, props_map)| {
+            props_map
+                .keys()
+                .map(move |qargs| (&self._gate_name_map[op], qargs.as_ref()))
+        })
+    }
+    /// Returns an iterator over the operation names in the target.
+    pub fn operation_names(&self) -> impl Iterator<Item = &String> {
+        self.gate_map.keys()
+    }
```

Comment:
`&String` is a bit weird I'd expect `&str` 
```suggestion
    pub fn operation_names(&self) -> impl Iterator<Item = &str> {
        self.gate_map.keys()
    }
```

But I'm not sure the compiler is smart enough to map that, so you might need:

```suggestion
    pub fn operation_names(&self) -> impl Iterator<Item = &str> {
        self.gate_map.keys().map(|x| x.as_str())
    }
```